### PR TITLE
use version.go file to store the actual version

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

don't create a VERSION file as it's unneeded and breaks DWCO build; instead bump version.go file to match actual reality and remove -next suffix

Change-Id: Ia97fa6347a525070bec1445102d6eace9632ca81
Signed-off-by: nickboldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/19526

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: make-release.sh [hunk 1]
```diff
@@ -107,24 +99,15 @@ else
...
-echo "${VERSION}" > VERSION
-
 QUAY_REPO="quay.io/devfile/devworkspace-controller:${VERSION}"
 docker build -t "${QUAY_REPO}" -f ./build/Dockerfile .
 docker push "${QUAY_REPO}"
 
 set -x
 bash -x ./deploy/generate-deployment.sh --use-defaults --default-image ${QUAY_REPO}
 
-# tag the release if the VERSION file has changed
```
Comment (mkuznyetsov) [inline make-release.sh:120]: Please do not remove this section, as generate-deployment.sh will generate templates, that we will need to commit

Comment (nickboldt) [reply make-release.sh:120]: oh, I thought this was only about hte VERSION file

File: make-release.sh [hunk 2]
```diff
@@ -30,14 +30,6 @@ bump_version () {
   BUMP_BRANCH=$2
 
   git checkout "${BUMP_BRANCH}"
-
-  echo "Updating project version to ${NEXT_VERSION}"
-  echo "${VERSION}" > VERSION
```
Comment (mkuznyetsov) [inline make-release.sh:35]: instead of VERSION file, we should echo/sed the version into version.go?

File: make-release.sh [hunk 3]
```diff
@@ -107,24 +99,15 @@ else
 fi
 set -e
 
-# change VERSION file
-echo "${VERSION}" > VERSION
```
Comment (mkuznyetsov) [inline make-release.sh:111]: same as https://github.com/devfile/devworkspace-operator/pull/378/files#r609660810

File: make-release.sh (hunk 2 above)
Comment (sleshchenko) [inline make-release.sh:35]: Don't we need to update version.go value here instead?

File: version/version.go [hunk 4]
```diff
@@ -13,5 +13,5 @@
 package version
 
 var (
-	Version = "1.0.0-next"
+	Version = "v0.3.0-next"
```
Comment (sleshchenko) [inline version/version.go:16]: I wonder if we should be fine with just `v0.3.0`, and then in https://github.com/devfile/devworkspace-operator/issues/377 we'll add git commit info.

**Review by sleshchenko — APPROVED** [review]

File: make-release.sh (hunk 2 above)
Comment (nickboldt) [reply make-release.sh:35]: yes!

Comment (nickboldt) [reply make-release.sh:35]: yes!

File: version/version.go (hunk 4 above)
Comment (nickboldt) [reply version/version.go:16]: does the `v` prefix break anything in go-speak version semantics?

**Review by sleshchenko — APPROVED** [review]

**Pull request merged**